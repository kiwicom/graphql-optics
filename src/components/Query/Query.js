import React from 'react'
import { withRouter } from 'react-router'
import Button from '../Button/Button'

const codeStyle = {
  fontFamily: 'monospace',
  whiteSpace: 'pre',
  backgroundColor: 'rgba(248, 248, 255, 1)',
  overflowX: 'auto',
  width: '100%',
}

const buttonStyle = {
  marginBottom: '20px',
}

const Query = ({ query, history, location, isActive }) => (
  <Button
    style={buttonStyle}
    isActive={isActive}
    onClick={() => history.push(`${location.pathname}/${query._id}`)}
  >
    <div style={codeStyle}>{query._source.graphql}</div>
  </Button>
)

export default withRouter(Query)
