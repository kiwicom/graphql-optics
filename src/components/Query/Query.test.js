import React from 'react'
import Query from './Query'
import { renderWithRouter } from '../../__tests__/helpers'

it('renders correctly', () => {
  const data = {
    _source: {
      graphql: '{allLocations(search: "Boulder")})',
    },
  }
  const tree = renderWithRouter({ children: <Query query={data} /> })
  expect(tree).toMatchSnapshot()
})
