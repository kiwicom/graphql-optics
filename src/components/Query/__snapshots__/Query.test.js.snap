// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="button"
  onClick={[Function]}
  style={
    Object {
      "marginBottom": "20px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "rgba(248, 248, 255, 1)",
        "fontFamily": "monospace",
        "overflowX": "auto",
        "whiteSpace": "pre",
        "width": "100%",
      }
    }
  >
    {allLocations(search: "Boulder")})
  </div>
</div>
`;
